### ISSUE TITLE
Adopt usage of `std::cell::LazyCell` and `std::sync::LazyLock` 

### ISSUE THREAD

**Issue by jorgehermo9** [vectordotdev/vector#20950]
In Rust 1.80 (https://blog.rust-lang.org/2024/07/25/Rust-1.80.0.html) `std::cell::LazyCell` and `std::sync::LazyLock` are finally stabilized and thus they can be used instead of the variants of `once_cell` crate. 

I think we could adopt the ones from the std and drop the usage of the `once_cell` crate once #20949 is merged.

We should also change this section in the `STYLE.md` file: https://github.com/vectordotdev/vector/blob/master/STYLE.md#global-state

**Comment by moderation** [comment]
Looks like there are quite a few fixes for `1.80.0` that need to occur. This is as far as I got using a conditional build of `cargo build --release --verbose --no-default-features --features api,sinks-axiom,sinks-console,sinks-http,sinks-prometheus,sources-file,sources-internal_metrics,sources-journald,sources-opentelemetry,transforms-filter`

```
error: unexpected `cfg` condition name: `tokio_unstable`
   --> lib/vector-buffers/src/lib.rs:142:11
    |
142 |     #[cfg(tokio_unstable)]
    |           ^^^^^^^^^^^^^^
    |
    = help: expected names are: `clippy`, `debug_assertions`, `doc`, `docsrs`, `doctest`, `feature`, `miri`, `overflow_checks`, `panic`, `proc_macro`, `relocation_model`, `rustfmt`, `sanitize`, `sanitizer_cfi_generalize_pointers`, `sanitizer_cfi_normalize_integers`, `target_abi`, `target_arch`, `target_endian`, `target_env`, `target_family`, `target_feature`, `target_has_atomic`, `target_has_atomic_equal_alignment`, `target_has_atomic_load_store`, `target_os`, `target_pointer_width`, `target_thread_local`, `target_vendor`, `test`, `ub_checks`, `unix`, and `windows`
    = help: consider using a Cargo feature instead
    = help: or consider adding in `Cargo.toml` the `check-cfg` lint config for the lint:
             [lints.rust]
             unexpected_cfgs = { level = "warn", check-cfg = ['cfg(tokio_unstable)'] }
    = help: or consider adding `println!("cargo::rustc-check-cfg=cfg(tokio_unstable)");` to the top of the `build.rs`
    = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration
note: the lint level is defined here
   --> lib/vector-buffers/src/lib.rs:6:9
    |
6   | #![deny(warnings)]
    |         ^^^^^^^^
    = note: `#[deny(unexpected_cfgs)]` implied by `#[deny(warnings)]`

error: unexpected `cfg` condition name: `tokio_unstable`
   --> lib/vector-buffers/src/lib.rs:148:15
    |
148 |     #[cfg(not(tokio_unstable))]
    |               ^^^^^^^^^^^^^^
    |
    = help: consider using a Cargo feature instead
    = help: or consider adding in `Cargo.toml` the `check-cfg` lint config for the lint:
             [lints.rust]
             unexpected_cfgs = { level = "warn", check-cfg = ['cfg(tokio_unstable)'] }
    = help: or consider adding `println!("cargo::rustc-check-cfg=cfg(tokio_unstable)");` to the top of the `build.rs`
    = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration
```
```
error: unexpected `cfg` condition value: `test`
   --> src/config/transform.rs:155:21
    |
155 |     #[cfg(any(test, feature = "test"))]
    |                     ^^^^^^^^^^^^^^^^
    |
    = note: expected values for `feature` are: `all-e2e-tests`, `all-integration-tests`, `all-logs`, `all-metrics`, `allocation-tracing`, `amqp-integration-tests`, `api`, `api-client`, `appsignal-integration-tests`, `aws-cloudwatch-logs-integration-tests`, `aws-cloudwatch-metrics-integration-tests`, `aws-config`, `aws-core`, `aws-ec2-metadata-integration-tests`, `aws-ecs-metrics-integration-tests`, `aws-integration-tests`, `aws-kinesis-firehose-integration-tests`, `aws-kinesis-streams-integration-tests`, `aws-s3-integration-tests`, `aws-sns-integration-tests`, `aws-sqs-integration-tests`, `axiom-integration-tests`, `azure-blob-integration-tests`, `azure-integration-tests`, `benches`, `chronicle-integration-tests`, `cli-tests`, `clickhouse-integration-tests`, `codecs-benches`, `codecs-syslog`, `component-validation-runner`, `component-validation-tests`, `databend-integration-tests`, `datadog-agent-integration-tests`, and `datadog-logs-integration-tests` and 207 more
    = help: consider adding `test` as a feature in `Cargo.toml`
    = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration
note: the lint level is defined here
   --> src/lib.rs:7:9
    |
7   | #![deny(warnings)]
    |         ^^^^^^^^
    = note: `#[deny(unexpected_cfgs)]` implied by `#[deny(warnings)]`

error: unexpected `cfg` condition value: `sources-http-client`
 --> src/internal_events/prelude.rs:5:5
  |
5 |     feature = "sources-http-client",
  |     ^^^^^^^^^^---------------------
  |               |
  |               help: there is a expected value with a similar name: `"sources-http_client"`
  |
  = note: expected values for `feature` are: `all-e2e-tests`, `all-integration-tests`, `all-logs`, `all-metrics`, `allocation-tracing`, `amqp-integration-tests`, `api`, `api-client`, `appsignal-integration-tests`, `aws-cloudwatch-logs-integration-tests`, `aws-cloudwatch-metrics-integration-tests`, `aws-config`, `aws-core`, `aws-ec2-metadata-integration-tests`, `aws-ecs-metrics-integration-tests`, `aws-integration-tests`, `aws-kinesis-firehose-integration-tests`, `aws-kinesis-streams-integration-tests`, `aws-s3-integration-tests`, `aws-sns-integration-tests`, `aws-sqs-integration-tests`, `axiom-integration-tests`, `azure-blob-integration-tests`, `azure-integration-tests`, `benches`, `chronicle-integration-tests`, `cli-tests`, `clickhouse-integration-tests`, `codecs-benches`, `codecs-syslog`, `component-validation-runner`, `component-validation-tests`, `databend-integration-tests`, `datadog-agent-integration-tests`, and `datadog-logs-integration-tests` and 207 more
  = help: consider adding `sources-http-client` as a feature in `Cargo.toml`
  = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration

error: unexpected `cfg` condition value: `sinks-sinks-aws_sns`
  --> src/sinks/mod.rs:23:38
   |
23 | #[cfg(any(feature = "sinks-aws_sqs", feature = "sinks-sinks-aws_sns"))]
   |                                      ^^^^^^^^^^---------------------
   |                                                |
   |                                                help: there is a expected value with a similar name: `"sinks-aws_sns"`
   |
   = note: expected values for `feature` are: `all-e2e-tests`, `all-integration-tests`, `all-logs`, `all-metrics`, `allocation-tracing`, `amqp-integration-tests`, `api`, `api-client`, `appsignal-integration-tests`, `aws-cloudwatch-logs-integration-tests`, `aws-cloudwatch-metrics-integration-tests`, `aws-config`, `aws-core`, `aws-ec2-metadata-integration-tests`, `aws-ecs-metrics-integration-tests`, `aws-integration-tests`, `aws-kinesis-firehose-integration-tests`, `aws-kinesis-streams-integration-tests`, `aws-s3-integration-tests`, `aws-sns-integration-tests`, `aws-sqs-integration-tests`, `axiom-integration-tests`, `azure-blob-integration-tests`, `azure-integration-tests`, `benches`, `chronicle-integration-tests`, `cli-tests`, `clickhouse-integration-tests`, `codecs-benches`, `codecs-syslog`, `component-validation-runner`, `component-validation-tests`, `databend-integration-tests`, `datadog-agent-integration-tests`, and `datadog-logs-integration-tests` and 207 more
   = help: consider adding `sinks-sinks-aws_sns` as a feature in `Cargo.toml`
   = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration

error: struct `TcpSocketError` is never constructed
  --> src/internal_events/tcp.rs:48:12
   |
48 | pub struct TcpSocketError<'a, E> {
   |            ^^^^^^^^^^^^^^
   |
   = note: `#[deny(dead_code)]` implied by `#[deny(warnings)]`

error: struct `UnixSocketError` is never constructed
  --> src/internal_events/unix.rs:36:12
   |
36 | pub struct UnixSocketError<'a, E> {
   |            ^^^^^^^^^^^^^^^

error: struct `UnixSocketSendError` is never constructed
  --> src/internal_events/unix.rs:61:12
   |
61 | pub struct UnixSocketSendError<'a, E> {
   |            ^^^^^^^^^^^^^^^^^^^

error: could not compile `vector` (lib) due to 6 previous errors
```

**Comment by jszwedko** [comment]
Rust 1.80.0 upgrade is happening here: https://github.com/vectordotdev/vector/pull/20949
